src/Features/LanguageServer/Protocol/Handler/Symbols/DocumentSymbolsHandler.cs

```diff
@@ -82,7 +81,7 @@ public override async Task<object[]> HandleRequestAsync(DocumentSymbolParams req
         private static SymbolInformation? GetSymbolInformation(
             RoslynNavigationBarItem item, Document document, SourceText text, string? containerName = null)
         {
-            if (item is not RoslynNavigationBarItem.SymbolItem symbolItem || symbolItem.SelectionSpan == null)
+            if (item is not RoslynNavigationBarItem.SymbolItem symbolItem || symbolItem.Location.InDocumentInfo == null)
```

Comment:
@dibarbet for confirmation.  FOr LSP i'm assuming it only supports document symbols navigating *within* the document.

Reply:
Interestingly it seems to depend on which type we return.  `SymbolInformation` defines its span as a `Location` which contains both the document and location inside the document.  But `DocumentSymbol` (the newer type) defines the spans as `Range` which does not not have a document piece.  So it seems like `SymbolInformation` does support navigation between documents, but `DocumentSymbol` does not.

Additionally, neither `DocumentSymbol` nor `SymbolInformation` allow multiple ranges, which we would definitely need for partial declarations.  I'm going to see if there are existing issues on these, otherwise I'll file them

Reply:
Feature request for allowing location in DocumentSymbol
https://github.com/microsoft/language-server-protocol/issues/1255

Reply:
lol.  i love the incosistencies.

Reply:
Feature request for multiple locations
https://github.com/microsoft/language-server-protocol/issues/1285